### ISSUE TITLE
Tekken: Dark Ressurection: save file invalid on MAC address change (needs warning)

### ISSUE THREAD

**Issue by Ac-Gamer** [hrydgard/ppsspp#15795]
### Game or games this happens in

ULUS10139

### What area of the game / PPSSPP

I tried playing Tekken: Dark Ressurection this morning when the screen said my game data corrupted. This isn't the first time this has happened, this also happened two months ago but I do not know what causes this issue or how to prevent i
![Screen Shot 2022-08-06 at 7 56 39 PM](https://user-images.githubusercontent.com/92767727/183249101-eb8754dc-2811-427c-927d-7e1df84e448a.png)
t

### What should happen

The game should load in the save data and run fine.

### Logs

none

### Platform

macOS

### Mobile phone model or graphics card

Intel HD Graphics 6000

### PPSSPP version affected

v1.12.3

### Last working version

None, downloaded the game 3 months ago in the same version

### Graphics backend (3D API)

Vulkan

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [ ] Try resetting settings or older versions and include if the issue is related.
- [X] Try without any cheats and without loading any save states.
- [X] Include logs or screenshots of issue.

**Comment by Ac-Gamer** [comment]
Another question, after my save data has "corrupted" is there any way to bring it back? or is it fully corrupted?

**Comment by Gamemulatorer** [comment]
Duplicate https://github.com/hrydgard/ppsspp/issues/13540?

**Comment by Ac-Gamer** [comment]
Not really, this is happening on a mac, and my ppsspp version is different version 1.12.13.

**Comment by Ac-Gamer** [comment]
I know this issue has been discussed many times before, but I still can't fully understand what causes this and how to prevent this issue. Has this been fixed in the latest 1.13.1 version of ppsspp? or is this something I have to fix myself?
This has happened 2 times now and is probably going to happen again sometime in the future.

**Comment by unknownbrackets** [comment]
The most common reasons for this are essentially from the game trying to prevent you from cheating, and you doing something it considers cheating.  Specifically:

1. Loading a save state intentionally doesn't revert savedata.  However, when saving the game may check that the savedata is exactly how it last left it.  If it isn't, CHEATER IDENTIFIED, and it marks the savedata corrupt.  It's actually not wrong, because using save states is a form of cheating.

2. Changing your MAC address in the settings.  On a real PSP, you can't change the MAC address so some games used this as a "PSP identifier."  If the MAC address is different, the game is sure you're trying to load the save state on a different PSP, which it considers cheating.

3. Running multiple instances of PPSSPP concurrently.  When you run a second instance of PPSSPP, it uses a different MAC address automatically, so this is essentially the same as the above.  Saves created with a second instance won't load with a first instance, and the opposite won't load either.

See #14937.  This is most likely 2 or 3 above, and that behavior is true of all platforms, macOS or otherwise.  Because it's game behavior, it's not really a thing that's going to change between PPSSPP versions either - PPSSPP is correctly running the game code which is correctly identifying a situation that it considers cheating.

-[Unknown]

**Comment by Ac-Gamer** [comment]
So this issue is most likely game specific? I don't see any reports of this in any other game either.

**Comment by Ac-Gamer** [comment]
I found it! i found what causes this issue (for me at least, probably also the case for other people who have encountered this issue) after some testing, I created a new savedata in tekken DR, saved it, then changed my MAC address, and when I can came back the screen said "game data corrupt." You were right. Almost everything surrounding this issue has something to do with the MAC address, and as you explained, the game specifically prevents you from cheating. This is why most of the time this is the only game that has this "game data corrupt" screen when launching; not many PSP games use this anti-cheat system.

Thank you so much, your assistance is greatly appreciated!
At least I now know what causes this bug. I remember that before each time this screen popped up when playing, I remember myself changing the MAC address on the settings for absolutely no reason :(
Its good that its identified what causes this issue and how to prevent it.
